Add initial support for Visual Studio 2026

jamesobutler opened the pull request:
See https://devblogs.microsoft.com/visualstudio/visual-studio-2026-insiders-is-here/

https://devblogs.microsoft.com/cppblog/c-language-updates-in-msvc-build-tools-v14-50/

Visual Studio 2026 in GitHub hosted runners progress at `https://github.com/actions/runner-images/issues/13035`

jamesobutler committed:
Add initial support for Visual Studio 2026

See https://devblogs.microsoft.com/visualstudio/visual-studio-2026-insiders-is-here/

https://devblogs.microsoft.com/cppblog/c-language-updates-in-msvc-build-tools-v14-50/

jcfr commented:
To help understand the failures:
* https://github.com/scikit-build/cmake-FindVcvars/pull/18

jamesobutler commented:
Testing: https://github.com/scikit-build/cmake-FindVcvars/actions/runs/17952851801/job/51056618203?pr=17
> 1: CMake Error at D:/a/cmake-FindVcvars/cmake-FindVcvars/src/FindVcvars.cmake:391 (cmake_host_system_information):
> 1:   cmake_host_system_information does not recognize <key> VS_18_DIR

`VS_18_DIR` is available in CMake as of this [commit](https://gitlab.kitware.com/cmake/cmake/-/commit/3392b371e2e039164f03435a5c7f70e003515cc60). It will be available in CMake 4.2.0 estimated Oct 1st release (https://gitlab.kitware.com/cmake/cmake/-/milestones/177#tab-issues). GitHub Action `jwlawson/actions-setup-cmake` doesn't appear to support pulling nightly builds of CMake, so will need to wait for the 4.2.0 release.

jamesobutler committed:
CI: Switch CMake setup action to lukka/get-cmake

https://github.com/lukka/get-cmake has support for CMake pre-releases which https://github.com/jwlawson/actions-setup-cmake does not support. It also appears that get-cmake action is actively maintained while actions-setup-cmake has not had an update in ~1.5 years.

jamesobutler commented:
I have been able to bypass the limitation of [`jwlawson/actions-setup-cmake`](https://github.com/jwlawson/actions-setup-cmake) not supporting CMake pre-releases by switching to [`lukka/get-cmake`](https://github.com/lukka/get-cmake). It also appears that `get-cmake` action is actively maintained while `actions-setup-cmake` has not had an update in ~1.5 years. `get-cmake` also currently has 120 stars on GitHub compared to `actions-setup-cmake` having 75 stars if that is any indication of the popularity and support of the project.

@jcfr This has allowed for CI to proceed successfully.

jamesobutler commented:
@jcfr This now uses CMake 4.2.0 stable for running the tests.

jcfr commented:
Thanks for your help working on this 🙏
